tests/TypedMessageTest.cc

```diff
@@ -0,0 +1,77 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include <gtest/gtest.h>
+#include <pulsar/Client.h>
+#include <pulsar/MessageBuilder.h>
+#include <pulsar/TypedMessage.h>
+#include <pulsar/TypedMessageBuilder.h>
+
+#include <vector>
+
+#include "IntSchema.h"
+
+using namespace pulsar;
+
+static std::string lookupUrl = "pulsar://localhost:6650";
+
+extern std::string unique_str();
+
+using IntMessageBuilder = TypedMessageBuilder<int>;
+
+TEST(TypedMessageTest, testIntValue) {
+    const auto topic = "typed-message-test-int-value-" + unique_str();
+    Client client(lookupUrl);
+    SchemaInfo schema(SchemaType::INT32, "", "");
+    Producer producer;
+    ASSERT_EQ(ResultOk, client.createProducer(topic, ProducerConfiguration().setSchema(schema), producer));
+    Consumer consumer;
+    ASSERT_EQ(ResultOk, client.subscribe(topic, "sub", ConsumerConfiguration().setSchema(schema), consumer));
+
+    std::vector<MessageId> messageIds;
+    constexpr int numMessages = 100;
+
+    IntSchema intSchema{schema};
+    for (int i = 0; i < numMessages; i++) {
+        MessageId messageId;
+        ASSERT_EQ(ResultOk, producer.send(intSchema.newMessage(i).build(), messageId));
```

Comment:
Using primitive types doesn't seem to show the benefits of this approach. Can we provide more examples for record types?

Reply:
It's a test for `encoder` and `decoder`, not an example for complicated cases. For example, the following implementations are also okay.

```c++
    std::function<std::string(int)> encoder() { return ""; }
    std::function<int(const char*, size_t)> decoder() { return 0; }
```

Reply:
I will redesign the tests soon. The consumer's API might also change.